gpytorch/lazy/diag_lazy_tensor.py

```diff
@@ -30,41 +27,8 @@ def __add__(self, other):
 
         return AddedDiagLazyTensor(other, self)
 
-    def _get_indices(self, left_indices, right_indices, *batch_indices):
```

Comment:
Let's verify removing this doesn't have speed implications, otherwise let's make a custom `_getitem` for `DiagLazyTensor`.

Reply:
Just about every LT (that we implement) should have a custom `_getitem` method and a custom `_expand_batch` method as part of this refactor. All LazyTensors *can* fall back on the default versions of these, but all LazyTensors should be able to easily support a custom method.

Plus, they'll be **MUCH** easier to write now, since you only have to handle two cases in `_getitem`